swift/extractor/infra/SwiftMangledName.h

```diff
@@ -25,28 +24,40 @@ struct SwiftMangledName {
   SwiftMangledName(SwiftMangledName&&) = default;
   SwiftMangledName& operator=(SwiftMangledName&&) = default;
 
+  template <typename... Args>
+  SwiftMangledName(Args&&... args) {
+    (operator<<(std::forward<Args>(args)), ...);
+  }
+
   // streaming labels or ints into a SwiftMangledName just appends them
   SwiftMangledName& operator<<(UntypedTrapLabel label) &;
   SwiftMangledName& operator<<(unsigned i) &;
 
+  template <typename Tag>
+  SwiftMangledName& operator<<(TrapLabel<Tag> label) & {
+    return operator<<(static_cast<UntypedTrapLabel>(label));
+  }
+
   // streaming string-like stuff will add a new part it only if strictly required, otherwise it will
   // append to the last part if it is a string
   template <typename T>
   SwiftMangledName& operator<<(T&& arg) & {
-    if (parts.empty() || !std::holds_alternative<std::string>(parts.back())) {
-      parts.emplace_back("");
-    }
-    std::get<std::string>(parts.back()) += std::forward<T>(arg);
+    value += arg;
```

Comment:
Optional: If we make this `absl::StrAppend(&value, arg);`, we can get rid of the `unsigned` overload.

Reply:
tried this, unfortunately while it gets rid of the `unsigned` overload, it requires two new overloads that do not work for `absl::StrAppend` while they do for `std::string::operator+=`: `char` and `llvm::StringRef`.

Reply:
Hmm... That's weird since `llvm::StringRef` has an [`operator std::string_view`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#ac06e4ec3286800a329c3202351a178ab).

Reply:
Yep, though I think using single-character string literals instead of `char`s wouldn't have been a problem; but let's keep it as is.